File: project/app/src/main/java/org/owntracks/android/ui/welcome/WelcomeMvvm.java
package org.owntracks.android.ui.welcome;

import org.owntracks.android.ui.base.view.MvvmView;

public interface WelcomeMvvm {

    // The view methods get called from all sorts of places. Fragments etc.
    interface View extends MvvmView {
        void showNextFragment();
        void setPagerIndicator(int position);

        // Called from Fragments to set button states
        void refreshNextDoneButtons();
    }
}

File: project/app/src/main/java/org/owntracks/android/ui/welcome/WelcomeMvvm.kt
package org.owntracks.android.ui.welcome

import org.owntracks.android.ui.base.view.MvvmView

interface WelcomeMvvm {
    // The view methods get called from all sorts of places. Fragments etc.
    interface View : MvvmView {
        fun setPagerIndicator(position: Int)
    }
}